server/src/main/java/org/elasticsearch/search/aggregations/bucket/sampler/random/RandomSamplerAggregator.java

```diff
@@ -116,6 +116,8 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
         if (scorer == null) {
             return LeafBucketCollector.NO_OP_COLLECTOR;
         }
+        sub.setScorer(scorer);
```

Comment:
The `probability >= 1.0` needs to be updated so that the collector overrides `setScorer` and passes that down to the sub values.